From 1cffffddfeaba44a4bb8809dae2caf07078745f3 Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Tue, 30 May 2017 12:32:52 +0100
Subject: [PATCH] No blue frame on toolbuttons

---
 kstyle/breezestyle.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kstyle/breezestyle.cpp b/kstyle/breezestyle.cpp
index ca5e2a29..022af36e 100644
--- a/kstyle/breezestyle.cpp
+++ b/kstyle/breezestyle.cpp
@@ -3484,7 +3484,7 @@ namespace Breeze
         const bool autoRaise( state & State_AutoRaise );
         const bool enabled( state & State_Enabled );
         const bool sunken( state & (State_On | State_Sunken) );
-        const bool mouseOver( enabled && (option->state & State_MouseOver) );
+        const bool mouseOver( enabled && (option->state & State_MouseOver) && false);
         const bool hasFocus( enabled && (option->state & (State_HasFocus | State_Sunken)) );
 
         /*
@@ -5866,7 +5866,7 @@ namespace Breeze
         // need to alter palette for focused buttons
         const State& state( option->state );
         const bool enabled( state & State_Enabled );
-        const bool mouseOver( enabled && (option->state & State_MouseOver) );
+        const bool mouseOver( enabled && (option->state & State_MouseOver) && false );
         const bool hasFocus( enabled && (option->state & State_HasFocus) );
         const bool sunken( state & (State_On | State_Sunken) );
         const bool flat( state & State_AutoRaise );
-- 
2.12.0

