diff --git a/lib/utils/x11helpers.cpp b/lib/utils/x11helpers.cpp
index 68ca9473..66352b44 100644
--- a/lib/utils/x11helpers.cpp
+++ b/lib/utils/x11helpers.cpp
@@ -75,9 +75,9 @@ qtcX11ShadowCreatePixmap(const QtCurve::Image *data)
 void
 qtcX11ShadowInit()
 {
-    int shadow_size = 30;
+    int shadow_size = 6;
     int shadow_radius = 4;
-    QtcColor c1 = {0.4, 0.4, 0.4};
+    QtcColor c1 = {0.65, 0.65, 0.65};
     QtcColor c2 = {0.2, 0.2, 0.2};
     QtCurve::Image *shadow_images[8];
     qtcShadowCreate(shadow_size, &c1, &c2, shadow_radius, false,
